src/uwtools/tests/test_cli.py

```diff
@@ -273,14 +273,14 @@ def test__dispatch_forecast(params):
 
 
 def test__dispatch_forecast_run():
-    args = ns()
+    args = ns(cycle="2023-01-01T00:00:00", config_file=1, dry_run=True, forecast_model="foo")
```

Comment:
I am running into some problems with using `datetime.date.fromisoformat` here. The docs indicate we might be in some trouble with older versions of Python:

> Changed in version 3.11: Previously, this method only supported the format YYYY-MM-DD.

Interestingly enough, this works in the test, but _NOT_ when I run the same thing from the command line. I do not know what's up with that!

Reply:
Do you mean that you have trouble getting `argparse` to use `datetime.date.fromisoformat` to check / parse the command-line value? Would it make any difference if we just let it be a `str` initially (from the perspective of `argparse`) and then parsed it ourselves after we had the value as a `str`?

Reply:
I feel like I sort of tried that by using a lambda. Let me give it another try to see what I can get to work with this thought.

Reply:
Luckily -- because I don't know how else I would have known this -- I'm using this same technique in the `iotaa-mpas` workflow, but like this:
```
>>> import datetime as dt
>>> dt.datetime.fromisoformat("2023-10-12T00")
datetime.datetime(2023, 10, 12, 0, 0)
```
No idea why `datetime` would behave differently from `date`...

Reply:
OMG. 🤦‍♀️